continuous_integration/environment-3.8-dev.yaml

```diff
@@ -19,9 +19,7 @@ dependencies:
   - zarr
   - tiledb-py
   - sqlalchemy
-  # Remove <2 version constraint once we've added pyarrow 2.0 compatibility
-  # xref https://github.com/dask/dask/issues/6754
-  - pyarrow>=0.14.0,<2
+  - pyarrow>=0.14.0,!=2.0.0
```

Comment:
Alternative could be to (temporarily) install the nightly wheel in this dev environment? (https://arrow.apache.org/docs/python/install.html#installing-nightly-packages) 
Then we would already have coverage for the new parquet dataset engine with pyarrow > 2.0 as well

Reply:
That's a good idea. I'm trying out CI with the nightly version of `pyarrow` over in https://github.com/dask/dask/pull/6936

Reply:
Another option could also be to skip the specific failing tests for pyrrow 2.0.0, but still install it in some CI environments (then at least we test the other (passing) tests with pyarrow 2.0)